application/F3DOptionsParser.cxx

```diff
@@ -675,6 +716,7 @@ void ConfigurationOptions::PrintReadersList()
 //----------------------------------------------------------------------------
 bool ConfigurationOptions::InitializeDictionaryFromConfigFile(const std::string& config)
 {
+  this->GlobalConfigEntries.clear();
   this->RegexConfigEntries.clear();
```

Comment:
why is this needed ?

Reply:
there was a single global block that would just get assigned https://github.com/f3d-app/f3d/blob/16fc29abfb227e76109ba6593d56f324921ee055/application/F3DOptionsParser.cxx#L786
now there's a list that gets appended to like the other one so I clear it too

Reply:
to be fair I fail to see why `this->RegexConfigEntries.clear();` is even needed. Unless I'm missing something `InitializeDictionaryFromConfigFile` is called only once ?

Reply:
Argument 1: Personally I would argue it makes sense to clear just in case. Or we could refactor the function so it creates both list within and returns them. Or we document to say it appends to the lists but does not clear them.
Either way just removing the clears "because we know" it's only called one seems sketchy?

Argument 2: ask this guy https://github.com/f3d-app/f3d/blame/9ac4f1c7c1723ebbd21f1afb9984f1fe56e7b3b5/application/F3DOptionsParser.cxx#L674 ;p

Reply:
Fair enough, maybe add a comment as to why we clear and that it is not absolutely needed  :)